d2common/d2fileformats/d2ds1/layer.go

```diff
@@ -0,0 +1,143 @@
+package d2ds1
+
+// layerStreamType represents a layer stream type
+type layerStreamType int
+
+// Layer stream types
+const (
+	layerStreamWall1 layerStreamType = iota
+	layerStreamWall2
+	layerStreamWall3
+	layerStreamWall4
+	layerStreamOrientation1
+	layerStreamOrientation2
+	layerStreamOrientation3
+	layerStreamOrientation4
+	layerStreamFloor1
+	layerStreamFloor2
+	layerStreamShadow1
+	layerStreamSubstitute1
+)
+
+type tileRow []Tile     // index is x coordinate
+type tileGrid []tileRow // index is y coordinate
+
+type grid interface {
+	Width() int
+	SetWidth(w int) grid
+
+	Height() int
+	SetHeight(h int) grid
+
+	Size() (w, h int)
+	SetSize(w, h int) grid
+
+	Tile(x, y int) *Tile
+	SetTile(x, y int, t *Tile)
+}
```

Comment:
is this interface even necessary? you use it only in SetWIdth/Height/Size methods, and you could return simply `*layer` there

Reply:
It isn't needed, I think this is just here because i used goland to write the boilerplate code from this interface, will remove.